### PR TITLE
Processing - Add Create, Edit, Delete method to the processing API

### PR DESCRIPTION
**Pull request by Fabrice-li**
* Add storage accessors
* Refactor some test code

### CODE REVIEW

File: src/Controller/Pia/ProcessingController.php [hunk 1]
```diff
@@ -73,4 +74,145 @@ public function listAction(Request $request)
...
+     *
+     * @Security("is_granted('CAN_CREATE_PROCESSING')")
+     *
+     * @return array
+     */
+    public function createAction(Request $request)
+    {
+        
+        $folder = $this->getResource($request->get('folder_id', -1), Folder::class);
+        $entity = $this->get('jms_serializer')->deserialize($request->getContent(), $this->getEntityClass(), 'json');
```
Comment (PapsOu) [inline src/Controller/Pia/ProcessingController.php]: Why unserializing the whole entity ?

Fetching only necessary fields `$request->get('name', null)` is not sufficient ?

Also please note that using the container will be deprecated soon (you should inject the service `jms_serializer` in the controller itself with type `JMS\Serializer\SerializerInterface`).

File: src/Controller/Pia/ProcessingController.php [hunk 2]
```diff
@@ -73,4 +74,145 @@ public function listAction(Request $request)
...
+     * @Security("is_granted('CAN_DELETE_PROCESSING')")
+     *
+     * @return array
+     */
+    public function deleteAction(Request $request, $id)
+    {
+        $processing = $this->getResource($id);
+        $this->canAccessResourceOr403($processing);
+
+        if (count($processing->getPias())) {
```
Comment (PapsOu) [inline src/Controller/Pia/ProcessingController.php]: Could you add the explicit check `if (count($processing->getPias() > 0)) {` ?

File: tests/api/130_ProcessingCest.php [hunk 3]
```diff
@@ -38,15 +53,88 @@ class ProcessingCest
...
+     * @depends create_processing_test
+     */
     public function list_processings_test(\ApiTester $I)
     {
         $I->amGoingTo('List available Processings');
 
         $I->login();
 
-        $I->sendGET('/processings');
+        $I->sendGET(ProcessingCest::ROUTE);
```
Comment (PapsOu) [inline tests/api/130_ProcessingCest.php]: Why not simply `self::ROUTE` ?

File: src/Controller/Pia/ProcessingController.php (hunk 1 above)
Comment (Fabrice-li) [reply src/Controller/Pia/ProcessingController.php]: You're right. I copied the method from RestController but it's not relevant here.

File: src/Controller/Pia/ProcessingController.php (hunk 2 above)
Comment (Fabrice-li) [reply src/Controller/Pia/ProcessingController.php]: Absolutely !

File: tests/api/130_ProcessingCest.php (hunk 3 above)
Comment (Fabrice-li) [reply tests/api/130_ProcessingCest.php]: Because we don't necessarily know what 'self' is outside the context. The class name is more explicit.

Comment (PapsOu) [reply tests/api/130_ProcessingCest.php]: Ok that explain the explicit visibility on the constant declaration.

**Pull request merged**